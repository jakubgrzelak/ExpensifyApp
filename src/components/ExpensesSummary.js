import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expesnsesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal}) => (
    <p>Viewing {expenseCount} {expenseCount === 1 ? 'expense' : 'expenses'} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</p>
);

const mapStateToProps = (state, props) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: expesnsesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);