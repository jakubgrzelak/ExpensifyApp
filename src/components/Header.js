import React from 'react'; 
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth.js';
import { Link } from 'react-router-dom';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/" exact={true}>
          <h1>Expensify</h1>
        </Link>
        <button className="login-button logut-button" onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
);
const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);
