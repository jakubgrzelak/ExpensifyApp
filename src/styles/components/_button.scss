.login-button {
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
    background-color: #008CBA;
    color: white;
    border: none;
    line-height: 1;
    text-decoration: none;
    
}

.logut-button {
    background: none;
}

.button--secondary {
    background: #888888;
}