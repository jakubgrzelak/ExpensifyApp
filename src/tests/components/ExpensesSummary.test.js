import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render Expenses summary page correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesTotal={2500} expenseCount={25}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should render correctly one expense on Expenses summary page', () => {
    const wrapper = shallow(<ExpensesSummary expensesTotal={250} expenseCount={1} />);
    expect(wrapper).toMatchSnapshot();
});
