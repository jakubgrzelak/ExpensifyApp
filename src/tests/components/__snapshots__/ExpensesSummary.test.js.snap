// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses summary page correctly 1`] = `
<p>
  Viewing 
  25
   
  expenses
   totalling 
  $25.00
</p>
`;

exports[`should render correctly one expense on Expenses summary page 1`] = `
<p>
  Viewing 
  1
   
  expense
   totalling 
  $2.50
</p>
`;
